libcrux-ml-kem/src/constant_time_ops.rs

```diff
@@ -78,6 +70,10 @@ pub(crate) fn compare_ciphertexts_in_constant_time(lhs: &[u8], rhs: &[u8]) -> u8
 }
 
 #[inline(never)] // Don't inline this to avoid that the compiler optimizes this out.
+#[cfg_attr(hax, hax_lib::requires(
+    lhs.len() == rhs.len() &&
+    lhs.len() == SHARED_SECRET_SIZE
+))]
```

Comment:
Here's my observation, with this pre-condition applied, `decapsulate` function in Libcrux_ml_kem.Ind_cca.fst takes 2.5 mins to lax-check. Without the pre-condition, that function takes ~1 min to lax-check. `encapsulate` function lax-checks right away in both cases.

Reply:
It seems F* has difficulty to have decapsulate function, is splitting it up an option?